add polyline object

GeneKao opened the pull request:
Added a polyline object similar to old viewer, @Licini @tomvanmele @brgcode code can be tested in `scripts/v120_polyline.py`
<img width="912" alt="Screenshot 2021-02-01 at 13 33 3" src="https://user-images.githubusercontent.com/1162842/106459775-a9b2c400-6492-11eb-954f-95a5083a4fe8.png">


### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

Licini commented on src/compas_view2/objects/polylineobject.py:
```diff
@@ -0,0 +1,67 @@
+        self._polylines = None
+        self.show_points = show_points
+        self.color_points = color_points
+        self.color_line = color_line
```
maybe call it `color_lines`?

Licini commented on src/compas_view2/objects/polylineobject.py:
```diff
@@ -0,0 +1,67 @@
+            'elements': make_index_buffer(list(flatten(elements))),
+            'n': len(positions) * 2
+        }
+        print()
```
empty print here

Licini reviewed (changes requested):
looks all good except the small things above

GeneKao commented:
@Licini fixed typos and added `width_lines` option for the line display width. 
@tomvanmele @brgcode Not sure if you guys have a plan to generalise this option from the viewer setting? Or do we add settings like the old viewer?

brgcode commented on scripts/v120_polyline.py:
```diff
@@ -19,6 +19,6 @@
 
 polyline = Polyline(pts)
-viewer.add(polyline, show_points=True, color_line=(1, 0, 0), line_width=5)
+viewer.add(polyline, show_points=True, color_lines=(1, 0, 0), width_lines=5)
```
we need to agree on better conventions for this. otherwise it is a bit of a guessing game as to what keywords arguments have to be used. in general, i noticed, for example, that python never uses underscores for this kind of things. it is also not the case for mpl...

GeneKao replied:
Maybe something like this? 

Point:
- pc = pt(s) color
- pw = pt(s) width

Line:
- lc = line(s) color
- lw = line(s) width

Face:

- fc = face(s) color
- fo = face(s) opacity

GeneKao commented:
@Licini @brgcode 
I just changed the name based on our agreed convention, please check if I understood it correctly. 😃 
Also added more options in the line object.

brgcode commented on src/compas_view2/objects/lineobject.py:
```diff
@@ -11,13 +11,21 @@ class LineObject(Object):
 
-    def __init__(self, data, name=None, is_selected=False, show_points=False, color_points=None, color_line=None):
+    def __init__(self, data, name=None, is_selected=False,
+                 show_point=False,
```
this needs to stay `points` with an `s` otherwise it is nonsensical...

brgcode commented on src/compas_view2/objects/polylineobject.py:
```diff
@@ -0,0 +1,79 @@
+    default_color_line = [0.4, 0.4, 0.4]
+
+    def __init__(self, data, name=None, is_selected=False,
+                 show_point=False,
```
`points` with an `s`
